### PR TITLE
Bear.py: Add more values for CAN_DETECT

### PR DESCRIPTION
**Pull request by Adrianzatreanu**

### CODE REVIEW

File: coalib/bears/Bear.py [hunk 1]
```diff
@@ -96,10 +96,11 @@ class Bear(Printer, LogPrinter):
     ...     INCLUDE_LOCAL_FILES = {'checkstyle.jar', 'google_checks.xml'}
 
     To keep track easier of what a bear can do, simply tell it to the CAN_FIX
-    and the CAN_DETECT sets. Possible values:
+    and the CAN_DETECT sets. Possible values but not limited to:
 
     >>> CAN_DETECT = {'Syntax', 'Formatting', 'Security', 'Complexity', 'Smell',
-    ... 'Redundancy', 'Simplification', 'Variable Misuse', 'Spelling', 'Other'}
+    ... 'Redundancy', 'Simplification', 'Variable Misuse', 'Spelling',
+    ... 'Unused Code', 'Memory Insufficiency', 'Unreachable Code', 'Design'}
```
Comment (sils1297) [inline coalib/bears/Bear.py]: CC @sims1253

Comment (sils1297) [inline coalib/bears/Bear.py]: isn't `Design` the same as `Smell`? I don't understand the memory thing fully.

Comment (Adrianzatreanu) [inline coalib/bears/Bear.py]: of course you don't, you python developer!

well i've come across bears which claim to tell you if your code wont compile due to memory insufficiency and where (such as segmentation faults in C probably) and theres no category i could fit them in

Comment (Adrianzatreanu) [inline coalib/bears/Bear.py]: i'm not 100% sure about design either, some claimed to improve the code design

Comment (sils1297) [inline coalib/bears/Bear.py]: :D let's use smell instead of design, they're the same IMO

**Comment by Adrianzatreanu** [comment]
moved here https://github.com/coala-analyzer/coala/pull/2374

**Pull request closed without merging**